src/server.rs

```diff
@@ -68,43 +125,43 @@ where
 
     fn call(&self, req: http::Request<A>) -> Self::Future {
         let is_gzip = is_gzip(&req);
-        Box::new(self.inner.call(req).map(move |rsp| {
+        trace!("GzipChunked: is_gzip={:?};", is_gzip);
+        let chunksz = self.chunk_size;
+        let compression = self.compression;
+
+        Box::new(self.inner.call(req).and_then(move |rsp| {
             let (mut parts, body) = rsp.into_parts();
+
             let body: Body = if is_gzip {
                 let mut encoder = GzEncoder::new(
-                    body.as_ref(),
-                    Compression::default());
-                parts.headers.insert(header::CONTENT_ENCODING, "gzip".parse().unwrap());
-                // let (mut tx_body, rx_body) = Body::channel();
-                // executor::spawn(future::lazy(move || {
-                //     let mut buf = [0u8; 4096];
-                //     loop {
-                //         match encoder.read(&mut buf) {
-                //             Ok(n) if n == 0 => { break; },
-                //             Ok(n) => {
-                //                 let chunk: hyper::Chunk = buf[0..n].to_vec().into();
-                //                 // while let Ok(Async::NotReady) = tx_body.poll_ready() { };
-                //                 tx_body.send_data(chunk)
-                //                     .expect("send data");
-                //             },
-                //             Err(e) => {
-                //                 eprintln!("error: {:?}", e);
-                //                 break;
-                //                 }
-                //         }
-                //     }
-                //     future::ok::<(),()>(())
-                // }));
-                let mut buf = Vec::new();
-                encoder.read_to_end(&mut buf);
-                buf.into()
+                    Cursor::new(Vec::<u8>::new()),
+                    compression,
+                );
+
+                let n = encoder.write(body.as_ref())
+                    .map_err(hyper::Error::Io)?;
+                let mut read = encoder.finish()
+                    .map_err(hyper::Error::Io)?;;
+                trace!("GzipChunked: encoded {:?} bytes", n);
+
+                read.set_position(0);
+                let stream = ChunkingStream {
+                    read,
+                    chunksz,
+                };
+
+                parts.headers.insert(
+                    header::CONTENT_ENCODING,
+                    "gzip".parse().unwrap()
+                );
+
+                Body::wrap_stream(stream)
```

Comment:
This function call makes me think you're using hyper master. We aren't using that yet in Conduit, and it will probably be a little while before we can...

Reply:
Thanks for pointing that out, will rewrite to use the release version of master --- I think I was looking at the wrong version of the docs when I wrote this.